tracing/span.go

```diff
@@ -199,7 +204,13 @@ func (s *Span) ShouldSample() bool {
 // It also sends the span into tracer, if the span should be sampled.
 // The context object passed in is used to control the timeout of the sending,
 // see Tracer.MaxRecordTimeout for more info.
+//
+// Returns ErrSpanAlreadyEnded if called more than once.
 func (s *Span) End(ctx context.Context, err error) error {
+	if !s.end.IsZero() {
+		return ErrSpanAlreadyEnded
```

Comment:
I don't have much experiencing with setting up tracing, I wonder if reporting an error could indeed prevent bugs on the calling side, couldn't it? If it doesn't create an actual harmful scenario than I'd also side on leaving it out otherwise it seems useful to catch misbehavior early on.